search-api/search-core/src/main/java/org/sunbird/search/processor/SearchProcessor.java

```diff
@@ -935,8 +935,7 @@ private QueryBuilder getSearchQuery(SearchDTO searchDTO) {
 	}
 
 	private QueryBuilder getQuery(SearchDTO searchDTO) {
-		return searchDTO.isFuzzySearch() ? prepareFilteredSearchQuery(searchDTO) : prepareSearchQuery(searchDTO);
-	}
-
+		return searchDTO.isFuzzySearch() ? (prepareSearchQuery(searchDTO) == null ?	prepareFilteredSearchQuery(searchDTO) : prepareSearchQuery(searchDTO)) : prepareSearchQuery(searchDTO);
```

Comment:
What are the chances of prepareSearchQuery becomes null ?
If prepareSearchQuery is not null - then the default SB behaviour of calling prepareFilteredSearchQuery won't be used in Sunbird environment. So this solution can not be proposed to Sunbird - we need better way of coding which is not causing any issues in Sunbird environment.